index.js

```diff
@@ -0,0 +1,257 @@
+        const newValue = txt.textContent;
+        txt.contentEditable = false;
+
+        if (newValue !== oldValue) {
+          const key = txt.parentElement.querySelector('.check').dataset.id;
+          const itemIndex = list.findIndex(item => item.index.toString() === key);
+          if (itemIndex !== -1) {
+            list[itemIndex].description = newValue;
+            save(list);
+          }
+        }
+      });
+    });
+  });
+}
+
+export function iconToggle() {
+  const menuItems = document.querySelectorAll('.menu');
+
+  menuItems.forEach((menuItem) => {
+    const deleteIcon = menuItem.nextElementSibling;
+
+    menuItem.addEventListener('click', (e) => {
+      const targetMenuItem = e.target;
+      const parentListItem = targetMenuItem.parentElement.parentElement;
+
+      parentListItem.classList.toggle('color');
+      targetMenuItem.classList.add('hide');
+      deleteIcon.classList.remove('hide');
+    });
+  });
+}
+
+import { iconToggle } from "./iconToggle";
+import { complete } from "./status";
+import { del } from "./remove";
+import { save } from "./save";
+import { edit } from "./edit";
+let index=0;
+let list=[];
+export const loadList=() =>{
+  const itemList=document.querySelector('.list-items');
+  const storedList = JSON.parse(localStorage.getItem('Todo' || []));
+  itemList.innerHTML = '';
+  if (storedList) {
+    list = storedList;
+    let listHtml = '';
+    list.forEach((item) => {
+      listHtml += `<li><input class='check' type="checkbox" data-id="${item.index}"><span class='txt'>${item.description}</span><span><i class="menu fa">&#xf142;</i><i data-id="${item.index}" class="delete hide fa">&#xf014;</i></span></li>`;
+      index=list.length;
+    })
+    itemList.innerHTML = listHtml;
+    save(list)
+   
+  }else {
+    list;
+    index;
+  }
+
+ iconToggle();
+ del(list);
+ complete(list);
+ edit(list);
+
+}
+
+export function refresh (){
+  const refreshBtn=document.querySelector('.refresh');
+  refreshBtn.addEventListener('click',()=>{
+    location.reload();
+  })
+}
+
+import { save } from "./save";
+
+export function del(list) {
+  const itemList = document.querySelector('.list-items');
+  itemList.addEventListener('click', function (e) {
+    e.preventDefault();
+    if (e.target.classList.contains('delete')) {
+      const listItem = e.target.closest('li');
+      if (listItem && itemList.contains(listItem)) {
+        itemList.removeChild(listItem);
+        const id = parseInt(e.target.getAttribute('data-id'));
+        const index = list.findIndex(item => item.index === id);
+        if (index !== -1) {
+          list[index].index--;
+          for (let i = index + 1; i < list.length; i++) {
+            list[i].index--;
+          }
+          list.splice(index, 1);
+          save(list);
+        }
+      }
+    }
+  });
+}
+
+export function save (list) {
+  localStorage.setItem('Todo', JSON.stringify(list));
+}
+
+import { save } from "./save";
+
+export function complete(list) {
```

Comment:
Great work! personally i felt this was overkill though and that doesnt follow the KISS rule try to keep it a bit simpler next time